src/shared/utils/fs.ts

```diff
@@ -216,7 +216,7 @@ export async function findProjectFolders(logger: Logger, roots: string[], exclud
 	const projectFolderPromises = allPossibleFolders.map(async (folder) => ({
 		exists: options && options.requirePubspec
 			? await hasPubspecAsync(folder)
-			: await hasPubspecAsync(folder) || await hasCreateTriggerFileAsync(folder) || await isFlutterRepoAsync(folder),
+			: options.forceFlutterWorkspace || await hasPubspecAsync(folder) || await hasCreateTriggerFileAsync(folder) || await isFlutterRepoAsync(folder),
```

Comment:
If `forceFlutterWorkspace` is set to `true` but `searchDepth` is not set to `1`, I think the results here would be unexpected (they'd include every folder at each of the `searchDepth` levels? If so, perhaps we should name this flag `onlyWorkspaceRoots` or similar and ignore `searchDepth`?